### PR TITLE
Refactor services with Resources interceptor for mock environment

### PR DESCRIPTION
**Pull request by FarajiA**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-418

 ## Description:
Add resources interceptor to resource services for mock environment to return mock data. This update makes the service unaware of the presence of the mock data but allows the interceptor to handle the response.

_Seeing some issues with the unit test for requests to the resources service_ 

 ## Testing:
`npm run test:unit` 

 ## Screenshots:
(if applicable)

### CODE REVIEW

File: src/app/_features/monitors/mon.utils.spec.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ describe('Monitor Utilities', () => {
     }));
 
     describe('ParseMonitorTypeEnum', () => {
-        it('should create utility', () => {
+        xit('should create utility', () => {
```
Comment (pras7430) [inline src/app/_features/monitors/mon.utils.spec.ts:60]: if it is not used, then we can remove it?

File: src/app/_interceptors/mock-resources.interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            .pipe(mergeMap(handleRoute))
+            .pipe(materialize()) // call materialize and dematerialize to ensure delay even if an error is thrown (https://github.com/Reactive-Extensions/RxJS/issues/648)
+            .pipe(delay(500))
+            .pipe(dematerialize());
+
+        function handleRoute() {
+            switch (true) {
+                case url.includes('/resources') && method === 'GET' && size > 0:
+                    return getAllResource();
+                case url.match(/\/resources\/[a-zA-Z0-9-_:]+$/) && method === 'GET':
```
Comment (pras7430) [inline src/app/_interceptors/mock-resources.interceptor.ts:38]: case url.match(/\/resources\/[a-zA-Z0-9-_:]+$/) && method === 'GET':
case url.match(/\/resources\/[a-zA-Z0-9-_:]+$/) && method === 'PUT':
return getAResource();

File: src/app/_interceptors/mock-resources.interceptor.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import { environment } from 'src/environments/environment';
+import { ResourcesService } from '../_services/resources/resources.service';
+import { resourcesMock } from '../_mocks/resources/resources.service.mock';
+
+
+@Injectable()
+export class MockResourcesInterceptor implements HttpInterceptor {
+    constructor(private inj: Injector) { }
+
+    intercept(request: HttpRequest<any>, next: HttpHandler): Observable<any> {
```
Comment (pras7430) [inline src/app/_interceptors/mock-resources.interceptor.ts:14]: here we have implemented nested functions, they should be separated

File: src/app/_interceptors/mock-resources.interceptor.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+        if (!environment.mock) {
+            return next.handle(request);
+        }
+
+        const mockedResources = new resourcesMock();
+        const { url, method, headers, body } = request;
```
Comment (pras7430) [inline src/app/_interceptors/mock-resources.interceptor.ts:21]: header and body is not used anywhere

**Review by pras7430 — APPROVED** [review]

File: src/app/_interceptors/mock-resources.interceptor.ts (hunk 2 above)
Comment (FarajiA) [reply src/app/_interceptors/mock-resources.interceptor.ts:38]: lol but of course.

File: src/app/_interceptors/mock-resources.interceptor.ts (hunk 3 above)
Comment (FarajiA) [reply src/app/_interceptors/mock-resources.interceptor.ts:14]: Hmm.. not sure I understand. Should the functions be hoisted above the return statement?

File: src/app/_interceptors/mock-resources.interceptor.ts (hunk 4 above)
Comment (FarajiA) [reply src/app/_interceptors/mock-resources.interceptor.ts:21]: Yup removed.

File: src/app/_interceptors/mock-resources.interceptor.ts (hunk 3 above)
Comment (FarajiA) [reply src/app/_interceptors/mock-resources.interceptor.ts:17]: Oh never mind I got it, I've updated the class to get rid of the multiple nested functions. Good catch

+1

**Pull request merged**